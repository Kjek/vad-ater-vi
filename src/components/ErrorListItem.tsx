interface ErrorListItemProps {
  restaurantHomeUrl?: string;
}

const ErrorListItem = (props: ErrorListItemProps) => {
  const { restaurantHomeUrl } = props;
  return (
    <>
      <li key='error-message'>
        <h3 className='text-2xl font-bold text-gray-800 dark:text-gray-300'>
          {
            'Fel vid hämtning av menyer gå till restaurangens hemsida istället: '
          }
        </h3>
        <a
          className='text-2xl font-bold text-blue-700 underline hover:no-underline dark:text-gray-500'
          href={restaurantHomeUrl}
        >
          {restaurantHomeUrl}
        </a>
      </li>
    </>
  );
};

export default ErrorListItem;
