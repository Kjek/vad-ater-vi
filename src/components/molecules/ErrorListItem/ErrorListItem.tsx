import Text from '@component/atoms/Text/Text';

interface ErrorListItemProps {
  restaurantHomeUrl?: string;
}

const ErrorListItem = ({ restaurantHomeUrl }: ErrorListItemProps) => {
  return (
    <>
      <li key='error-message'>
        <Text variant='h3'>
          {
            'Fel vid hämtning av menyer gå till restaurangens hemsida istället: '
          }
        </Text>
        <a
          className='text-2xl font-bold text-blue-700 underline hover:no-underline dark:text-gray-500'
          href={restaurantHomeUrl}
        >
          {restaurantHomeUrl}
        </a>
      </li>
    </>
  );
};

export default ErrorListItem;
