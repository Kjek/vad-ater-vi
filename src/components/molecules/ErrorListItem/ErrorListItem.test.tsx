import { render } from '@testing-library/react';
import { describe, expect, it } from 'vitest';
import ErrorListItem from './ErrorListItem';

describe(ErrorListItem, () => {
  it('renders correctly', () => {
    const { container } = render(
      <ErrorListItem restaurantHomeUrl='www.example.com' />
    );
    expect(container).toBeInTheDocument();
    expect(container).toBeVisible();
  });

  it('matches snapshot', () => {
    const defaultRender = render(
      <ErrorListItem restaurantHomeUrl='www.example.com' />
    );
    expect(defaultRender).toMatchSnapshot();
  });
});
