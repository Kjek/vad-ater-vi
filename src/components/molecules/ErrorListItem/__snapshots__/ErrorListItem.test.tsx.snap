// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ErrorListItem > matches snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <li>
        <h3
          class="text-2xl font-bold text-gray-800 dark:text-gray-300"
        >
          Fel vid hämtning av menyer gå till restaurangens hemsida istället: 
        </h3>
        <a
          class="text-2xl font-bold text-blue-700 underline hover:no-underline dark:text-gray-500"
          href="www.example.com"
        >
          www.example.com
        </a>
      </li>
    </div>
  </body>,
  "container": <div>
    <li>
      <h3
        class="text-2xl font-bold text-gray-800 dark:text-gray-300"
      >
        Fel vid hämtning av menyer gå till restaurangens hemsida istället: 
      </h3>
      <a
        class="text-2xl font-bold text-blue-700 underline hover:no-underline dark:text-gray-500"
        href="www.example.com"
      >
        www.example.com
      </a>
    </li>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
